desktop/cmp/grid/GridModel.js

```diff
@@ -264,10 +264,11 @@ export class GridModel {
         return [...this.columns];
     }
 
-    /** @param {Column[]} cols */
+    /** {(HoistColumn[]|Object[])} cols - Columns, or configs to create them. */
     @action
     setColumns(cols) {
-        this.columns = [...cols];
+        this.columns = cols.map(c => c instanceof Column ? c : new Column(c));
+        this.validateColumns();
```

Comment:
Can we refactor this to validate *before* mutating the columns.  Think it makes more sense, and it allows a caller of this model to recover from a mistake without wrecking the state of a healthy GridModel.

Just refactor the validate method to take the proposed new columns.